### ISSUE TITLE
Empty (NA) Sample Names break read_cell_seg_data

### ISSUE THREAD

**Issue by gadenbuie** [PerkinElmer/phenoptr#9]
I've just run into a cell segmentation file that has a final row that is missing all fields except for `Path`. I'm not sure how this extra row ended up in this particular file, but it causes the following lines to throw an error when `length(unique(df[[sample_name]]))` is greater than 1 but one of those values is `NA`.

https://github.com/PerkinElmer/phenoptr/blob/71d9d6cbc747e5f0ec5dbf1a4b965dcd461c7d3d/R/read_cell_seg_data.R#L94-L97

The error that is returned in this case is hard to parse.

```
Error in if (.min == .max) return(x) : 
  missing value where TRUE/FALSE needed
```

The error is actually thrown from `remove_common_prefix()`. I have [a fix in this branch](https://github.com/gadenbuie/phenoptr/tree/fix-na-sample-name) that both requires more than 1 non-missing unique Sample Name for the above `if` block and also allows `NA`s to propagate through `remove_common_prefix()`. 

If you'd like a PR I'll be happy to submit it. I updated the tests and the tests in `test_read_cell_seg_data.R` all pass. Other tests fail but I don't think it's related to this.

**Comment by gadenbuie** [comment]
I do agree with you in general. But on the other hand, I'd like to be able to fix this after importing, inside R. I'm importing a large number of files at once so this error breaks my pipeline. Also because `read_cell_seg_csv()` errors out, I can't get the data into R unless I either manually reproduce the steps in that function or edit the source data (which I'd strongly prefer not do).

I understand throwing the error but it would be nice to allow the user to decide how they would like to handle it, which I believe could be accomplished with an appropriate warning or message (while still allowing `read_cell_seg` to complete).

OTOH -- are you certain that inForm will never write out a cell seg file with a missing value in the `Sample Name` column? (Note: I'm genuinely curious as knowing this will help me with QC.)

**Comment by PKI-Kent** [comment]
I don't think inForm should ever be missing a value for Sample Name. This is the name of the im3 file from which the cell data is derived so it should always be available. It sounds like your problematic file may have been truncated somehow, is that possible?

**Comment by gadenbuie** [comment]
I see, thanks for the answer. I completely agree then, this should definitely given an error rather than allowing broken data through. It might be useful to add a check with an informative error message

```r
if (any(is.na(df[[sample_name]]))) {
  stop("Column ", paste0("`", sample_name, "`"), 
       " contains missing values. Check input data file.")
}
```

**Comment by AB-Kent** [comment]
Addressed in https://github.com/akoyabio/phenoptr/commit/d1e4466eb6e736e1636e02f7cdcb563ceb2611b2
